components/icon/icon.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <component
+    :is="icon"
+    data-qa="dt-icon"
+    :class="iconSize"
+  />
```

Comment:
Just realized we should probably have the [title](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title) as a prop. Since this needs to be the first child within the SVG it makes it a bit awkward to set in this component. 

Probably need to use js to add the title element on mounted and then a watch to update the dom if the prop changes.
Other option is to add a title prop into the actual icon vue components on the dialtone side

Reply:
I'd prefer to add it to dialtone, it's easier to do it that way.